### PR TITLE
Update to libswoc 1.3.8

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
Update libswoc to [1.3.8](https://github.com/SolidWallOfCode/libswoc/releases/tag/1.3.8). This has a number of minor improvements and introduces the `Vectray` class.

### CODE REVIEW

**Review by bneradt — COMMENTED** [review]
Looks fine. Looks like RAT wants some licensing headers update. Is this just a port in from the libswoc version I reviewed a week or so ago? Anything interesting I should focus on?

**Comment by SolidWallOfCode** [comment]
I'm unclear on why RAT is failing now, and not when this was originally added. My approach is to exclude "swoc" from RAT as is done for YAML-CPP. We might want to check with the ASF if the libswoc style is acceptable, which AFAIK is the case.

It's the same version of libswoc you reviewed earlier.

**Comment by bneradt** [comment]
OK, I reproduced this locally. This is kind of funny. We check the report like so:

https://github.com/apache/trafficserver-ci/blob/main/jenkins/github/rat.pipeline#L57

On master, that grep produces this:

```
$ grep '0 Unknown Licenses' RAT.txt
30 Unknown Licenses
```

See, it passes. The report has literally the string `0 Unknown Licenses`. Unfortunately there's a 3 in front of that.

With your patch in this PR, @SolidWallOfCode , this report now shows 31 Unknown Licenses:

```
31 Unknown Licenses
```

Therefore the grep fails.

Clearly we need to update CI to check more precisely. In the meantime, I'm not sure what to do about these licenses. Do we have to officially vet this with someone in Apache?

**Comment by SolidWallOfCode** [comment]
We're treating libswoc as a third party library and so we can exclude it from the checks as we do with YAML-CPP. However we may want to update our RAT checking. If you look [here](https://www.apache.org/licenses/LICENSE-2.0) you can see the ASF supports the SPDX notation used in libswoc.

**Comment by bneradt** [comment]
> However we may want to update our RAT checking. 

I don't think updating is a simple thing for us. We simply use Apache's https://github.com/apache/creadur-rat. I was hoping the latest version would support the short license format, but I do not see any updates to that which would accept shortened SPDX URLs.

File: ci/rat-regex.txt [hunk 1]
```diff
@@ -67,4 +67,5 @@ port\.h
 ^catch[.]hpp$
 ^configuru.hpp$
 ^yamlcpp$
+^swoc$
```
Comment (bneradt) [inline ci/rat-regex.txt:70]: Since that's what we do with yamlcpp files, I suppose this fine.

**Review by bneradt — APPROVED** [review]

**Review by brbzull0 — APPROVED** [review]
Looks good.

**Pull request merged**